cinema/views.py

```diff
@@ -1 +1,65 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Movie, Genre, Actor, CinemaHall, MovieSession
+from cinema.serializers import (
+    GenreSerializer,
+    MovieSerializer,
+    ActorSerializer,
+    CinemaHallSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieListSerializer,
+    MovieRetrieveSerializer,
+    MovieSessionRetrieveSerializer
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
```

Comment:
Consider adding `prefetch_related` for 'genres' and 'actors' directly in the `queryset` definition if these fields are frequently accessed, even outside of the 'list' action. This can improve performance by reducing the number of database queries.